assets/js/blocks/product-new/block.js

```diff
@@ -56,6 +58,20 @@ class ProductNewestBlock extends Component {
 						}
 					/>
 				</PanelBody>
+				{ getSetting( 'hide_out_of_stock' ) !== 'yes' && (
```

Comment:
These bits have been removed from all similar components.

Filtering the 'Out of stock' status at ProductStockControl level, rather than in the parent components. This means the filters can still be applied on WC stores with 'hide_out_of_stock' switched on.